src/components/TeacherComponents/OnboardingForm/onboardingActions/collectOnboardingTrackingProps.ts

```diff
@@ -0,0 +1,144 @@
+import { useUser } from "@clerk/nextjs";
+
+import { OnboardingFormProps } from "../OnboardingForm.schema";
+
+import {
+  onboardingRoleOptions,
+  OnboardingRoleValue,
+} from "./onboardingRoleOptions";
+
+import {
+  UserAccountVerificationStatusValueType,
+  UserOnboardingCompletedProperties,
+  UserOnboardingProgressedProperties,
+  UserRoleTypeValueType,
+} from "@/browser-lib/avo/Avo";
+import { UserResource } from "clerk";
+
+/* Reported for manually entered schools or as a fallback when the URN cannot be extracted
+from the school picker data */
+const DUMMY_URN = "000000";
+
+/**
+ * Produces onboarding tracking properties from the user and onboarding form
+ */
+export function collectOnboardingTrackingProps(
+  distinctUserId: string,
+  user: NonNullable<ReturnType<typeof useUser>["user"]>,
+  data: OnboardingFormProps,
+): UserOnboardingCompletedProperties & UserOnboardingProgressedProperties {
+  const signUpDate = (user.createdAt ?? new Date())?.toISOString().slice(0, 10);
+  const userDetailsLastModifiedDate = (user.updatedAt ?? new Date())
+    .toISOString()
+    .slice(0, 10);
+
+  return {
+    platform: "owa",
+    product: "user account management",
+    engagementIntent: "explore",
+    componentType: "continue_button",
+    eventVersion: "2.0.0",
+    analyticsUseCase: "Teacher",
+    userId_: distinctUserId,
```

Comment:
Continue to identify the user with their posthog id. Failing to do this causes users to be thrown out of the `use-auth-owa` feature flag at the start of onboarding.